src/frametree.cpp

```diff
@@ -475,8 +475,10 @@ bool FrameTree::focusInDirection(Direction direction, bool externalOnly)
     shared_ptr<HSFrame> neighbour = curframe->neighbour(direction);
     if (neighbour) { // if neighbour was found
         shared_ptr<HSFrameSplit> parent = neighbour->getParent();
-        // alter focus (from 0 to 1, from 1 to 0)
-        parent->swapSelection();
+        if (parent) {
+            // alter focus (from 0 to 1, from 1 to 0)
+            parent->swapSelection();
+        }
         return true;
```

Comment:
Is this correct even in the case where !parent? Or should the return true be inside the if(parent) block?

Reply:
If it has a neighbour, the neighbour must have a parent. So the only purpose of `if (parent)` is to silence the compiler warning.

Reply:
Ok, just wanted to be sure. Thanks for your explanation!